### PR TITLE
Box Component

### PR DESCRIPTION
**Pull request by bearni95**
## Reason:

Closes ["Box Component"](https://app.asana.com/0/1204649083736950/1204549680288790/f)

## Changes made (preferably with images/screenshots):
- Primitive layout component that gives access to design tokens for margin

## Check off the following:

- [x] I have reviewed my changes and run the appropriate tests.
- [ ] I have have run `rush change` to add the appropriate change logs.
- [x] I have added/edited docs.
- [x] I have added tutorials.
- [x] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: packages/libs/react-ui/src/components/Card/Card.css.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const container = style([
 export const fullWidthClass = style([sprinkles({ width: '100%' })]);
 
 export const stackClass = style([
-  sprinkles({ marginY: 0 }),
+  sprinkles({ marginY: '$0' }),
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Card/Card.css.ts:24]: I don't believe 0 needs the `$` prefix since it is a not a custom value

**Review by eileenmguo — CHANGES_REQUESTED** [review]
I think there may have been some miscommunication about this component. Where are you getting the styles for this component from?

**Comment by bearni95** [comment]
As discussed, tried integrating a similar pattern to the one vanilla-extract uses for their own `<Box>` component. 

PR reverted to draft due to the experimental status of the current implementation

**Review by eileenmguo — APPROVED** [review]
This is super cool! I think it works great :)

File: packages/libs/react-ui/src/components/Box/Box.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  marginX = undefined,
+  marginY = undefined,
+  marginTop = undefined,
+  marginBottom = undefined,
+  marginLeft = undefined,
+  marginRight = undefined,
+  top = undefined,
+  bottom = undefined,
+  left = undefined,
+  right = undefined,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Box/Box.tsx:22]: What is the reason for adding these?

File: packages/libs/react-ui/src/components/Box/Box.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import type { Meta, StoryObj } from '@storybook/react';
+import React from 'react';
+
+const selectOptions: (keyof typeof vars.sizes | undefined)[] = [
+  undefined,
+  ...(Object.keys(vars.sizes) as (keyof typeof vars.sizes)[]),
+];
+
+const meta: Meta<IBoxProps> = {
+  title: 'Layout/Vanilla Extract Box',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Box/Box.stories.tsx:15]: I would also change this to Layout/Box

File: packages/libs/react-ui/src/components/Box/Box.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { Sprinkles, sprinkles } from '../../styles';
+
+import React, { createElement, ElementType } from 'react';
+
+export interface IBoxProps extends Sprinkles {
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Box/Box.tsx:5]: Can we try making the type more strict? So there is a typescript error if anyone tries to pass in anything other than the margin props

File: packages/libs/react-ui/src/components/Box/Box.tsx (hunk 2 above)
Comment (bearni95) [reply packages/libs/react-ui/src/components/Box/Box.tsx:22]: Leftovers from testing, removed now

**Review by eileenmguo — APPROVED** [review]
Super excited about this one!

**Pull request merged**